### PR TITLE
feat(streaming): split chunks in lookup executor

### PR DESCRIPTION
**Pull request by pleiadesian**
Signed-off-by: Zulai Wang <zl31wang@gmail.com>

## What's changed and what's your intention?

As title. Also, `take()` is introduced in array builders to enable tracking current chunk size and outputing chunks every time window size is reached.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

## Refer to a related PR or issue link (optional)
#1891

### CODE REVIEW

File: src/common/src/array/decimal_array.rs [hunk 1]
```diff
@@ -166,6 +168,16 @@ impl ArrayBuilder for DecimalArrayBuilder {
             data: self.data,
         })
     }
+
+    fn take(&mut self) -> Result<DecimalArray> {
```
Comment (skyzh) [inline src/common/src/array/decimal_array.rs:172]: I'd prefer to create a new array builder every time we need to build a new array. Having such function in array builder could be confusing for future developers.

File: src/stream/src/lib.rs [hunk 2]
```diff
@@ -39,6 +39,7 @@
 #![feature(proc_macro_hygiene)]
 #![feature(stmt_expr_attributes)]
 #![feature(unzip_option)]
+#![feature(iterator_try_collect)]
```
Comment (skyzh) [inline src/stream/src/lib.rs:42]: `itertools` already have `try_collect`.

**Review by skyzh — COMMENTED** [review]
Rest LGTM

File: src/stream/src/common/builder.rs [hunk 3]
```diff
@@ -57,11 +68,43 @@ impl StreamChunkBuilder {
...
+    /// the capacity limit
+    ///
+    /// A [`StreamChunk`] will be returned when `size == capacity`
+    pub fn append_row_with_limit(
+        &mut self,
+        op: Op,
+        row_update: &RowRef<'_>,
+        row_matched: &Row,
+    ) -> Result<Option<StreamChunk>> {
+        self.ops.push(op);
```
Comment (skyzh) [inline src/stream/src/common/builder.rs:89]: Data chunks must not split update-insert and update-delete apart, so this doesn't seem like an ultimate solution for split chunk. We can revisit this part later.

**Review by skyzh — APPROVED** [review]

File: src/stream/src/common/builder.rs (hunk 3 above)
Comment (BugenZhao) [reply src/stream/src/common/builder.rs:89]: This may lead to unstable e2e tests. 🤤

**Pull request merged**